Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -589,18 +586,96 @@ PlatformImagePtr ImageDecoderCG::createFrameImageAtIndex(size_t index, Subsampli
     return maskedImage ? maskedImage : image;
 }
 
+String ImageDecoderCG::decodeUTI(const SharedBuffer& data) const
+{
+    auto uti = String(CGImageSourceGetType(m_nativeDecoder.get()));
+    if (uti != "public.heif"_s && uti != "public.heic"_s)
+        return uti;
+
+    if (data.size() < sizeof(unsigned)) {
+        ASSERT_NOT_REACHED();
+        return uti;
+    }
+
+    static constexpr auto ftypSignature = FourCC("ftyp");
+    static constexpr auto avifBrand = FourCC("avif");
+    static constexpr auto avisBrand = FourCC("avis");
+    
+    auto boxUnsigned = [&data](unsigned index) -> unsigned {
+        static constexpr bool littleEndian = false;
+        const unsigned* boxBytes = reinterpret_cast<const unsigned*>(data.data());
+        // Let's make all the numbers in big endian because FourCC are big endian.
+        return flipBytesIfLittleEndian(boxBytes[index], littleEndian);
```

Comment:
Passing a constant false to a conditional function doesn’t make a lot of sense

Reply:
This is for clarity only. Because the function takes a bool, I thought passing a constant `false` will not be clear.